tsdb/querier.go

```diff
@@ -123,6 +127,27 @@ func (q *querier) Select(ms ...*labels.Matcher) (SeriesSet, error) {
 	return NewMergedSeriesSet(ss), nil
 }
 
+func (q *querier) Exemplars(exemplarStorage ExemplarStorage, ms ...*labels.Matcher) ([][]exemplar.Exemplar, error) {
+	var exemplars [][]exemplar.Exemplar
+	ss, err := q.Select(ms...)
+	if err != nil {
+		return nil, err
+	}
+	// ss = ss.(baseChunkSeries)
+	for ss.Next() {
+		hash := ss.At().Labels().Hash()
+		seriesExemplars, err := exemplarStorage.Select(hash)
+		if err != nil { // failure to find exemplars for a single series in the seriesset shouldn't be fatal
```

Comment:
All local storage failures should fail the query, if this has happened then something has gone very very wrong.

Reply:
That depends on what we decide to return errors for from `exemplarStorage.Select`. `ss` is the series we found that match the label matcher for the query. That doesn't necessarily mean there's exemplars stored for every one of those stored.

Reply:
Returning no data when no data exists is not an error.